pkg/internal/util/cli/spinner.go

```diff
@@ -60,35 +62,49 @@ var _ io.Writer = &Spinner{}
 // NewSpinner initializes and returns a new Spinner that will write to w
 func NewSpinner(w io.Writer) *Spinner {
 	return &Spinner{
-		stop:   make(chan struct{}, 1),
-		ticker: time.NewTicker(time.Millisecond * 100),
-		mu:     &sync.Mutex{},
-		writer: w,
+		stop:    make(chan struct{}, 1),
+		stopped: make(chan struct{}, 1),
+		mu:      &sync.Mutex{},
+		writer:  w,
 	}
 }
 
 // SetPrefix sets the prefix to print before the spinner
 func (s *Spinner) SetPrefix(prefix string) {
 	s.mu.Lock()
+	defer s.mu.Unlock()
 	s.prefix = prefix
-	s.mu.Unlock()
 }
 
 // SetSuffix sets the suffix to print after the spinner
 func (s *Spinner) SetSuffix(suffix string) {
 	s.mu.Lock()
+	defer s.mu.Unlock()
 	s.suffix = suffix
-	s.mu.Unlock()
 }
 
 // Start starts the spinner running
 func (s *Spinner) Start() {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+	// don't start if we've already started, also flag that we've started
+	if s.running {
+		return
+	}
+	s.running = true
+	// start / create a frame ticker
+	s.ticker = time.NewTicker(time.Millisecond * 100)
```

Comment:
should we add a timeout?
```	go func() {
		time.Sleep(10 * time.Second)
		s.stop <- struct{}{}
	}()
```

Reply:
I'm not sure why we'd do this? It should spin as long as kind is still busy doing something which may take arbitrarily long (e.g. pulling an image)

Reply:
just wondering if can we guarantee that won't ever be in an endless loop? :sweat_smile:

Reply:
the process exits regardless (it's in a background task) :-)

callers should defer s.Stop() (we do), similar to a server or other background process

Reply:
s.Stop() is only called from (s *Status) End() , right?

Reply:
yeah, the same as Start() is only called from *Status